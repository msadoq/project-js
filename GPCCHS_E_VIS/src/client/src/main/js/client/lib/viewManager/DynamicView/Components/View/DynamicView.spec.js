import _ from 'lodash';
import {
  isDroppedDataValid,
} from './DynamicView';

const invalidEP = {
  comObjects: [
    'ReportingParameter',
  ],
  item: 'AIV_TM_HK9018P069',
  nameSpace: 'SDB',
  sessionName: '600',
};

const invalidEP2 = {
  catalogName: 'Reporting',
  item: 'AIV_TM_HK9018P069',
  nameSpace: 'SDB',
  sessionName: '600',
};

const singleReportingEP = {
  catalogName: 'Reporting',
  comObjects: [
    'ReportingParameter',
  ],
  item: 'AIV_TM_HK9018P069',
  nameSpace: 'SDB',
  sessionName: '600',
};

const clcwPacketEP = {
  catalogName: 'TelemetryPacket',
  comObjects: [
    'CLCW',
    'DecommutedPacket',
    'IsisAggregation',
    'RM',
    'TM',
  ],
  item: 'AIV_TM_HK9000',
  nameSpace: 'SDB',
  sessionName: '600',
};

const decommutedPacket = {
  catalogName: 'TelemetryPacket',
  comObjects: [
    'CLCW',
    'DecommutedPacket',
    'IsisAggregation',
    'RM',
    'TM',
  ],
  nameSpace: 'SDB',
  sessionName: '600',
};

describe('viewManager/DynamicView/Components/View/DynamicView', () => {
  describe('isDroppedDataValid', () => {
    _.map([undefined, '', null, 'lqskdj', invalidEP, invalidEP2], (test) => {
      it(`should be invalid to drop wrong data, or malformed EP - ${test}`, () => {
        expect(isDroppedDataValid(test)).toBe(false);
      });
    });
    _.map([singleReportingEP, clcwPacketEP, decommutedPacket], (test) => {
      it(`should be valid to drop well validated EP - ${test}`, () => {
        expect(isDroppedDataValid(test)).toBe(true);
      });
    });
  });
});
