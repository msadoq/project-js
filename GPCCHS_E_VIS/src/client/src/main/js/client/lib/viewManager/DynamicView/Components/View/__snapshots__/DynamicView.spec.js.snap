// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`viewManager/DynamicView/Components/View/DynamicView buildTable should match the snapshot 1`] = `
<table
  className="table table-striped table-bordered table-condensed table-hover"
>
  <thead>
    <tr>
      <th
        className="text-center"
      >
        name
      </th>
      <th
        className="text-center"
      >
        converted value
      </th>
      <th
        className="text-center"
      >
        raw value
      </th>
      <th
        className="text-center"
      >
        extracted value
      </th>
      <th
        className="text-center"
      >
        validity state
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        GENE_AM_CCSDSVERS1
      </td>
      <td>
        2
      </td>
      <td>
        2
      </td>
      <td>
        0c 0c 0c 0c 0c 0c 0c 0c 0c 0c
      </td>
      <td>
        INVALID
      </td>
    </tr>
    <tr>
      <td>
        GENE_AM_CCSDSVERS2
      </td>
      <td>
        0
      </td>
      <td>
        0
      </td>
      <td>
        01 01 01 01 01 01 01 01 01 01
      </td>
      <td>
        INVALID
      </td>
    </tr>
  </tbody>
</table>
`;
